hw3/01-routing.js

```diff
@@ -34,16 +34,52 @@ const server = http.createServer((req, res) => {
     return result;
   };
 
-  if (req.url === '/') {
+  if (req.url === "/") {
     let routeResults = getRoutes();
 
-    res.writeHead(200, { 'Content-Type': 'text/html' });
+    res.writeHead(200, { "Content-Type": "text/html" });
     res.write(`<h1>Exercise 01</h1>`);
     res.write(`<ul> ${routeResults} </ul>`);
-    res.end();
+    return res.end();
   }
 
   // Add your code here
+  else if (req.url === "/welcome") {
+    res.writeHead(200, { "Content-Type": "text/html" });
+    res.write(
+      `<h1>Welcome to the wolcome page! As always, you are welcome.</h1>`
+    );
+    return res.end();
+  } else if (req.url === "/redirect") {
+    res.writeHead(302, { Location: "/redirected" });
+    return res.end();
+  } else if (req.url === "/redirected") {
+    res.writeHead(200, { "Content-Type": "text/html" });
+    res.write(`<h1>You've been redirected.</h1>`);
+    return res.end();
+  } else if (req.url === "/cache") {
+    res.writeHead(200, {
+      "Content-Type": "text/plain",
+      "Cache-Control": "max-age=86400",
+    });
+    res.write(`this resource was cached`);
+    return res.end();
+  } else if (req.url === "/cookie") {
+    res.writeHead(200, {
+      "Content-Type": "text/plain",
+      "Set-Cookie": "hello=world",
+    });
+    res.write(`cookies... yummm`);
+    return res.end();
+  } else if (req.url === "/check-cookies") {
+    res.writeHead(200, { "Content-Type": "text/plain" });
+    res.write(`${req.headers.cookie.includes("hello") ? "yes" : "no"}`);
+    return res.end();
+  } else {
+    res.writeHead(404, { "Content-Type": "text/html" });
+    res.write(`<h1>404 - page not found</h1>`);
+    return res.end();
+  }
 });
 
```

Comment:
Everything looks goods but just a coding recommendation would be adding switch-case instead of if else.